### ISSUE TITLE
Related resources don't obtain object via the resource_uri's resource

### ISSUE THREAD

**Issue by timdawborn** [django-tastypie/django-tastypie#949]
tl;dr: The type of object that a resource_uri points to is not asserted to be the same as the type of the foreign key.

```python
class FooResource(ModelResource):
  class Meta:
    resource_name = 'foo'
    queryset = Foo.objects.all()

class BarResource(ModelResource):
  class Meta:
    resource_name = 'bar'
    queryset = Bar.objects.all()

class BazResource(ModelResource):
  foo = ForeignKey(FooResource, 'foo')

  class Meta:
    always_return_data = True
    authentication = Authentication()
    authorization = Authorization()
    list_allowed_methods = ('post',)
    resource_name = 'baz'
    queryset = Baz.objects.all()
  ...
```

When assigning to the `foo` attribute of `BazResource` via a resource_uri, you can pass any resource_uri as the value (not necessarily a resource_uri to the `FooResource`), as long as it has the same non-resource-name kwargs as `FooResource`. This comes down to the implementation of `get_via_uri`:

```python
      def get_via_uri(self, uri, request=None):
        ...
        try:
            view, args, kwargs = resolve(chomped_uri)
        except Resolver404:
            raise NotFound(u"The URL provided '%s' was not a link to a valid resource." % uri)

        bundle = self.build_bundle(request=request)
        return self.obj_get(bundle=bundle, **self.remove_api_resource_names(kwargs))
```

If I supply `/api/bar/1/` as the value of the `foo` attribute to `BazResource`, the kwargs are extracted as `{'pk': '1', 'api_name': 'api', 'resource_name': 'bar'}`. However, `obj_get` is then called on `self`, which is a `FooResource` instance. At no point is the resource name verified to be correct.

```bash
$ curl -v -X POST -H 'Content-Type: application/json' --data '{"foo": "/api/bar/1/"}' http://localhost:8000/api/baz/
* About to connect() to localhost port 8000 (#0)
*   Trying 127.0.0.1...
* connected
* Connected to localhost (127.0.0.1) port 8000 (#0)
> POST /api/baz/ HTTP/1.1
> User-Agent: curl/7.24.0 (x86_64-apple-darwin12.0) libcurl/7.24.0 OpenSSL/0.9.8r zlib/1.2.5
> Host: localhost:8000
> Accept: */*
> Content-Type: application/json
> Content-Length: 22
>
* upload completely sent off: 22 out of 22 bytes
* HTTP 1.0, assume close after body
< HTTP/1.0 201 CREATED
< Date: Fri, 31 May 2013 05:28:07 GMT
< Server: WSGIServer/0.1 Python/2.7.2
< Vary: Accept, Accept-Language, Cookie
< Content-Type: application/json
< Location: http://localhost:8000/api/baz/25/
< Content-Language: en
<
* Closing connection #0
{"foo": "/api/foo/1/", "resource_uri": "/api/baz/25/"}
```

Is this by design or is it a bug? I can't see anything anywhere obvious that's indicating that this is by design.

**Comment by SeanHayes** [comment]
I added some more tests, I think this has been fixed.
